src/nodevisitor.rs

```diff
@@ -48,6 +65,14 @@ impl<'a> StaticPathParser<'a> {
         visitor.visit(&node);
         Ok(visitor)
     }
+
+    pub fn get_path_info(path: &'a str) -> Result<PathInfoFlags, String> {
+        let parser = Self::check(path)?;
+        match parser.valid {
+            VisitStatus::Valid => Ok(PathInfoFlags::STATIC | PathInfoFlags::DEFINED_ORDER),
+            _ => Ok(PathInfoFlags::INVALID),
```

Comment:
Why not error here? With some relevant error message? What the INVALID means?

Reply:
We return an error on path compilation failure (from `check` method)
`INVALID` was needed before `pathParse` could fail.
I will rename it now to `NONE` to mark some initial value.
Anyway, no need to expose the flags now, since there are dedicated APIs (`pathIsStatic`, `pathHasDefinedOrder`)